code/controllers/subsystem/shuttle.dm

```diff
@@ -594,6 +594,15 @@ SUBSYSTEM_DEF(shuttle)
 			return TRUE
 
 /datum/controller/subsystem/shuttle/proc/get_containing_shuttle(atom/A)
-	for(var/obj/docking_port/mobile/M in mobile)
-		if(M.is_in_shuttle_bounds(A))
-			return M
+	var/list/mobile_cache = mobile
+	for(var/i in 1 to mobile_cache)
+		var/obj/docking_port/port = mobile_cache[1]
+		if(port.is_in_shuttle_bounds(A))
+			return port
+
+/datum/controller/subsystem/shuttle/proc/get_containing_dock(atom/A)
+	var/list/stationary_cache = stationary
+	for(var/i in 1 to stationary_cache)
```

Comment:
Shouldn't this be 1 to stationary_cache.len?

Reply:
Why do both of these as numerical for loops when you do not even use the number except to get the list element?

Reply:
`for(var/i in list)` creates a copy of the list and iterates through the copy.
`for(var/i in 1 to list.len)` lets you iterate directly through the list that already exists

Reply:
Why does that even matter for such a small list? Why is every for loop not done like this?

Reply:
Most lists should use this but most people aren't in the habit of writing it this way. You also can't do typed loops this way when you need to filter the list.

Also we only realized this was a possible performance improvement a couple years ago and we don't really enforce it